airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -89,11 +105,9 @@ public ConnectorSpecification spec() throws IOException {
 
   @Override
   public AirbyteConnectionStatus check(JsonNode config) {
-    try (final Database database = createDatabase(config)) {
-      // attempt to get current schema. this is a cheap query to sanity check that we can connect to the
-      // database. `currentSchema()` is a jooq method that will run the appropriate query based on which
-      // database it is connected to.
-      database.query(this::getCurrentDatabaseName);
+    try (final JdbcDatabase database = createDatabase(config)) {
+      // attempt to get metadata from the database as a cheap way of seeing if we can connect.
+      database.query(conn -> JdbcUtils.resultSetToJson(conn.getMetaData().getCatalogs()));
```

Comment:
This test doesn't depend on the number of "objects" in the db right? (tables/schemas... ?) Just want to make sure we won't hit a edge case with someone with 1000s of objects that will fail the test.

Reply:
nope. `getCatalogs` should be equivalent to the number of databases is a database instance.